Automattic/wp-calypso
# Media: Update to use Redux instead of media library selected items store

#### Changes proposed in this Pull Request

* Post Editor: Use Redux for selected media in the editor media modal
* Media: Use Redux for selected media in Media section
* Gutenberg: Use Redux for selected media in Calypsoify
* Site Settings: Use Redux for selected media in Podcast cover image and Site icon

#### Testing instructions

* Checkout this branch.
* Thoroughly test the media for regressions in the following areas (browsing, managing/editing, uploading, deleting), and compare with production:
  * Classic post editor
    * Make sure to try both HTML and Visual tabs
  * Site media library
  * Gutenberg editor - Calypsoify
  * Site settings - Podcast cover image and Site Icon settings.
  * In all the above, make sure you try drag&drop upload functionality.
  * In all the above, make sure you try different media sources.
* Verify all tests pass.

Note: #40272 precedes this PR and should be merged first. Feel free to review this one separately, though.

Note: This PR is part of an iteration to remove the media library selected items store, see #40269 for the full effort that removes it completely. Also, see #39379 for the complete media reduxification effort.
